### PR TITLE
chore: upgrade Azure/go-autorest/autorest to v0.11.1

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Upgrade `Azure/go-autorest/autorest` to v0.11.1 to solve the issue https://github.com/kubernetes-sigs/azuredisk-csi-driver/issues/441

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
Use the following commands to upgrade.
```
hack/pin-dependency.sh github.com/Azure/go-autorest/autorest v0.11.1
hack/pin-dependency.sh github.com/Azure/go-autorest/autorest/adal v0.9.0
hack/pin-dependency.sh github.com/Azure/go-autorest/autorest/date v0.3.0
hack/pin-dependency.sh github.com/Azure/go-autorest/autorest/mocks v0.4.0
hack/pin-dependency.sh github.com/Azure/go-autorest/logger v0.2.0
hack/pin-dependency.sh github.com/Azure/go-autorest/tracing v0.6.0
hack/update-vendor.sh
hack/lint-dependencies.sh
```

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

/kind bug
/assign @andyzhangx @feiskyer 
/priority important-soon
/sig cloud-provider
/area provider/azure

### CODE REVIEW

**Comment by dims** [comment]
/assign @liggitt 

new LICENSE files in a few dirs but this is just a existing dependency update. Looks like they ended up creating a number of small go modules in the same repo.

**Review by feiskyer — APPROVED** [review]
/lgtm
/retest

File: go.mod
Comment (liggitt) [inline go.mod:14]: for reference: v0.9.6 was equivalent to v13.4.0+incompatible

File: go.mod
Comment (liggitt) [inline go.mod:155]: This loopback dependency (where a nested module depends on a parent component that is not a module with a module-incompatible version tag) is ... interesting. I'm not sure what issues this could cause in the future.

Looking closer, I can't actually figure out why this dependency was added... there's no code in the base repo, and all subpackages declare their own modules... seems like this could just be dropped as a dependency by the `github.com/Azure/go-autorest/*` modules

**Comment by liggitt** [comment]
/approve
/lgtm
/hold
mechanics look fine, need to pass windows jobs before this merges

File: go.mod
Comment (liggitt) [reply go.mod:155]: weird, looks like this was added in https://github.com/Azure/go-autorest/pull/528 ... I'm still not sure why having all the submodules reference a parent containing no code is desired

**Comment by feiskyer** [comment]
all jobs are green now
/hold cancel

**Comment by liggitt** [comment]
this will be moved back into the milestone as part of the [phased reopen for 1.20 merges](https://docs.google.com/document/d/1yI7o7R2bOB7QtkHUFMHMb5XObiJmzIAnJPuxhdAX-CA/edit)

**Pull request merged**
